pmlc/dialect/pxa/transforms/autotile.h

```diff
@@ -74,14 +74,18 @@ class UnionGenerator<Single> {
 // ArrayRef<int64_t>) to a double, which is 'inf' for infeasible tilings.  For
 // example:
 
-inline double DummyCostModel(ArrayRef<int64_t> tile) { return 1.0; }
+inline double DummyCostModel(mlir::AffineParallelOp op, ArrayRef<int64_t> tile, void* args) {
+  return 1.0;
+}
 
 // Given a generator and cost model, find the best tile size, return empty
 // tiling when all tiles are infeasible
 template <typename Generator, typename CostModel>
-llvm::SmallVector<int64_t, 8> findBestTileSize(const Generator &generator,
+llvm::SmallVector<int64_t, 8> findBestTileSize(mlir::AffineParallelOp op,
+                                               const Generator &generator,
                                                const CostModel &costModel,
-                                               ArrayRef<int64_t> ranges) {
+                                               ArrayRef<int64_t> ranges,
+                                               void *args) {
```

Comment:
I'd like to avoid using `void*` for context, what do you want to pass here? Couldn't we use a lambda that captures any required context? Or even a functor object?

Reply:
args is for the target info (memory size, #processors...). Do you have any better idea?

Reply:
Yes, the thing that plugs in the cost model could contain all this extra info. Either as captures to a lambda or as a Functor object.

Reply:
I want to pass a structure including the above info. However, the structures may differ for different Autotile examples.